### PR TITLE
Set user agent for Angular, React, Vue libs - plus Jest

### PR DESCRIPTION
**Pull request by robertjd**
Uses the new property in AuthJS to append the framework lib to the user agent

Also adds unit testing via Jest for React, Vue

Angular not yet tested, will add a "unit" test to app.component.spec.ts after #143 lands in master

### CODE REVIEW

File: packages/okta-vue/package.json [hunk 1]
```diff
@@ -8,15 +8,33 @@
     "src"
   ],
   "scripts": {
+    "prebuild": "npm run build:package-info",
     "prestart": "npm run build",
     "pretest": "npm run build && npm run build:harness",
     "prepublish": "npm run build",
-    "test": "npm run --prefix test/e2e/harness/ test",
+    "jest": "jest src/",
+    "test": "npm run jest && npm run --prefix test/e2e/harness/ test",
```
Comment (jmelberg-okta) [inline packages/okta-vue/package.json]: I'm leaning towards thinking we should prepend `npm run lint` to this command. That way we can start enforcing the `eslint` style.

File: util/write-package-info.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+const workingDirectory = process.argv[2];
+const destinationFile = process.argv[3];
+const packageJsonPath = path.join(process.cwd(), workingDirectory, 'package.json')
+const configDest = path.join(process.cwd(), workingDirectory, destinationFile);
+const packageJson = require(packageJsonPath);
+const packageInfo = {
+  name: packageJson.name,
+  version: packageJson.version
+};
+const output = 'export default ' + JSON.stringify(packageInfo, null, 2) + ';'
```
Comment (jmelberg-okta) [inline util/write-package-info.js]: nit: There is no `;` in our `okta-vue/src/*.js` files.

File: util/write-package-info.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const fs = require('fs');
+const path = require('path');
+
+const workingDirectory = process.argv[2];
+const destinationFile = process.argv[3];
+const packageJsonPath = path.join(process.cwd(), workingDirectory, 'package.json')
```
Comment (jmelberg-okta) [inline util/write-package-info.js]: Missing semicolon

File: packages/okta-vue/package.json [hunk 4]
```diff
@@ -8,15 +8,33 @@
...
     "start": "npm run --prefix test/e2e/harness/ start",
     "build": "rimraf dist/ && cross-env NODE_ENV=production webpack --config webpack.config.js --output-library-target=umd -p",
     "build:harness": "npm --prefix test/e2e/harness install",
+    "build:package-info": "node ../../util/write-package-info.js . src/packageInfo.js",
```
Comment (jmelberg-okta) [inline packages/okta-vue/package.json]: We should consider using [`lerna run`](https://github.com/lerna/lerna#run) to handle this for us.

Comment (robertjd) [reply packages/okta-vue/package.json]: This task needs to be integrated into the build pipeline of each package, and ran locally while working on that specific package, so I'm not sure how lerna would help here?

File: util/write-package-info.js (hunk 2 above)
Comment (robertjd) [reply util/write-package-info.js]: This util script is at the root, so not part of lerna.  I can put together a PR to add eslint to the root, but we'd have to do testing to make sure it doesn't break what we just fixed for windows.

File: util/write-package-info.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+const workingDirectory = process.argv[2];
+const destinationFile = process.argv[3];
+const packageJsonPath = path.join(process.cwd(), workingDirectory, 'package.json');
+const configDest = path.join(process.cwd(), workingDirectory, destinationFile);
+const packageJson = require(packageJsonPath);
+const packageInfo = {
+  name: packageJson.name,
+  version: packageJson.version
+};
+const output = 'export default ' + JSON.stringify(packageInfo, null, 2) + ';';
```
Comment (jmelberg-okta) [inline util/write-package-info.js]: As noted earlier, the trailing `+ ';'` will result in eslint issues in our Vue package. Can we make this conditional?

Comment (robertjd) [reply util/write-package-info.js]: I excluded this file using the eslintignore in the vue package

Comment (jmelberg-okta) [reply util/write-package-info.js]: Really? I'm seeing this after a clean install:
screen shot 2018-03-19 at 9 23 04 am

**Pull request merged**